code/obj/machinery/manufacturer.dm

```diff
@@ -285,6 +287,30 @@
 		dat += "<br>"
 		dat += build_material_list()
 
+		dat += "<HR>"
+
+		dat +="<B>Scanned Card:</B> <A href='?src=\ref[src];card=1'>([src.scan])</A><BR>"
+		if(scan)
+			var/datum/data/record/account = null
+			account = FindBankAccountByName(src.scan.registered)
+			if (account)
+				dat+="<B>Current Funds</B>: [account.fields["current_money"]] Credits<br>"
+		dat+= src.temp
+		dat+="<HR>"
+
+		dat += "<B>Ores Available for Purchase:</B><br><small>"
+		for(var/obj/machinery/ore_cloud_storage_container/S in by_type[/obj/machinery/ore_cloud_storage_container])
+			if(S.broken)
+				continue
+			dat += "<B>[S.name] at [get_area(S)]:</B><br>"
+			var/list/ores = S.ores
+			for(var/ore in ores)
+				var/datum/ore_cloud_data/OCD = ores[ore]
+				if(!OCD.for_sale || !OCD.amount)
+					continue
+				var/taxes = round(max(rockbox_globals.rockbox_client_fee_min,abs(OCD.price*rockbox_globals.rockbox_client_fee_pct/100)),0.01) //transaction taxes for the station budget
+				dat += "[ore]: [OCD.amount] ($[OCD.price+taxes+(!rockbox_globals.rockbox_premium_purchased ? rockbox_globals.rockbox_standard_fee : 0)]/ore) (<A href='?src=\ref[src];purchase=1;storage=\ref[S];ore=[ore]'>Purchase</A>)<br>"
+
 		dat += "</small><HR>"
```

Comment:
also optimize string operations (ideally big multiline string where possible, list + `Join` where not)

Reply:
the Manufacturer code actually already does that list join thing :pelterfrog:

fixed up some of my additions to be multiline, but there wasn't actually much to change. Resolved in pending commit.